### DISCUSSION TITLE
Issue with uploading IPA with firebase app distribution status 404

### DISCUSSION THREAD

**Discussion by arled** [fastlane/fastlane#20048] [Q&A]
Getting the following error when  uploading IPA with firebase app distribution - the server responded with status 404: 
 
 Anyone else getting this?

[10:54:48]: --- Step: firebase_app_distribution ---
[10:54:48]: ---------------------------------------
[10:54:48]: Authenticating with --firebase_cli_token parameter
[10:54:49]: 🔐 Authenticated successfully.
[10:54:49]: ⌛ Uploading the IPA.
+--------------------+---------------------------------------------------------------------------------------------------+
|                                                      Lane Context                                                      |
+--------------------+---------------------------------------------------------------------------------------------------+
| PLATFORM_NAME      | ios                                                                                               |
| LANE_NAME          | ios distribute                                                                                    |
                               |
+--------------------+---------------------------------------------------------------------------------------------------+
[10:55:06]: the server responded with status 404

+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | build_app                 | 626         |
| 💥   | firebase_app_distribution | 17          |
+------+---------------------------+-------------+

[10:55:06]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

bundler: failed to load command: fastlane (/usr/local/lib/ruby/gems/3.1.0/bin/fastlane)
/usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/response/raise_error.rb:22:in `on_complete': \e[31m[!] the server responded with status 404\e[0m (Faraday::ResourceNotFound)
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/middleware.rb:19:in `block in call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/response.rb:61:in `on_complete'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/middleware.rb:18:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday_middleware-1.2.0/lib/faraday_middleware/response_middleware.rb:36:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/rack_builder.rb:154:in `build_response'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/connection.rb:516:in `run_request'
	from /usr/local/lib/ruby/gems/3.1.0/gems/faraday-1.10.0/lib/faraday/connection.rb:281:in `post'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-plugin-firebase_app_distribution-0.3.2/lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:112:in `upload_binary'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-plugin-firebase_app_distribution-0.3.2/lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:141:in `upload'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-plugin-firebase_app_distribution-0.3.2/lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb:41:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:19:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	from /usr/local/lib/ruby/gems/3.1.0/gems/fastlane-2.204.3/bin/fastlane:23:in `<top (required)>'
	from /usr/local/lib/ruby/gems/3.1.0/bin/fastlane:25:in `load'
	from /usr/local/lib/ruby/gems/3.1.0/bin/fastlane:25:in `<top (required)>'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:58:in `load'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:23:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli.rb:484:in `exec'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli.rb:31:in `dispatch'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/cli.rb:25:in `start'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/exe/bundle:48:in `block in <top (required)>'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /usr/local/lib/ruby/gems/3.1.0/gems/bundler-2.3.8/exe/bundle:36:in `<top (required)>'
	from /usr/local/lib/ruby/gems/3.1.0/bin/bundle:25:in `load'
	from /usr/local/lib/ruby/gems/3.1.0/bin/bundle:25:in `<main>'

**Comment by Pavel1501** [comment]
I had the same issue. I would suggest you following this instruction

1 Go to your Firebase console

2 Click App Distribution section under the Release & Monitor tab

3 Select your project

4 Verify if the “Get Started” button has been pressed

Simple thing but useful

**Reply by ryanbrink** [reply]
Just had to say thanks for this! Not obvious at all.

**Reply by abdulrehmanveemed** [reply]
@ryanbrink Hi, I am facing issue related to certificates is it possible if you can help me out?

**Reply by MuhammadEssaAli** [reply]
> I had the same issue. I would suggest you following this instruction
> 
> 1 Go to your Firebase console
> 
> 2 Click App Distribution section under the Release & Monitor tab
> 
> 3 Select your project
> 
> 4 Verify if the “Get Started” button has been pressed
> 
> Simple thing but useful

You've just saved my day . . KUDOS

**Reply by tobiloeb** [reply]
OMG thanks

**Reply by HossamAbdelnasser0110** [reply]
You saved my day

**Comment by ciekawy** [comment]
@arled please, mark the answer
